src/EditorFeatures/Core/Snippets/RoslynLSPSnippetExpander.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Reflection;
+using System.Text;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.LanguageServer;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+
+namespace Microsoft.CodeAnalysis.Snippets
+{
+    [Export(typeof(IRoslynLSPSnippetExpander))]
+    [Export(typeof(RoslynLSPSnippetExpander))]
```

Comment:
Why export as both?

Reply:
We need the AbstractSnippetCompletionProvider to mef import the IRoslynLSPSnippetExpander which only has the CanExpandSnippet bit since the ExpandSnippet in the RoslynLSPSnippetExpander has stuff not availble on the features layer